index.js

```diff
@@ -138,47 +174,69 @@ function nowServing(line) {
 const line = []
 
 // ***** Scope & Closures - Question 1 *****
-
+let i = 0
+let a = 1
+function takeATicketNumber(line) {
+  line.push(a)
+  i++
+  a++
+  return `Welcome. You are ticketNumber${i}`
+}
+// im probably doing this wrong, only way I came up with
+// to pass tests
 
 // *** Uncomment the lines below to test
-// console.log("%cScope & Closures - Question 1", "color: red")
+console.log("%cScope & Closures - Question 1", "color: red")
 
-// console.log(takeATicketNumber(line))
-// // => `Welcome. You are ticket number 1`
+console.log(takeATicketNumber(line))
+// => `Welcome. You are ticket number 1`
 
-// console.log(takeATicketNumber(line))
-// // => `Welcome. You are ticket number 2`
+console.log(takeATicketNumber(line))
+// => `Welcome. You are ticket number 2`
 
-// console.log(nowServing(line))
-// // => `Currently serving 1.`
+console.log(nowServing(line))
+// => `Currently serving 1.`
 
-// console.log(nowServing(line))
-// // => `Currently serving 2.`
+console.log(nowServing(line))
+// => `Currently serving 2.`
 
-// console.log(takeATicketNumber(line))
-// // => `Welcome. You are ticket number 3`
+console.log(takeATicketNumber(line))
+// => `Welcome. You are ticket number 3`
 
 // console.log("%c----------", "color: red") 
 
 // ***** Scope & Closures - Question 2 *****
+function ticketNumberGeneratorFunc() {
+  let i = 0
+  let a = 1
+  return function takeATicketNumber(line) {
+    line.push(a)
+    i++
+    a++
+    return `Welcome. You are ticketNumber${i}`
+  }
+}
```

Comment:
This is a tricky, advanced one! Nice work getting it! A higher order function is a function that accepts and/or returns another function. So this works, too, but can be simplified similarly to the previous question
```suggestion
function ticketNumberGeneratorFunc() {
  let i = 0
  return function takeATicketNumber(line) {
     i++
    line.push(i)
    return `Welcome. You are ticketNumber${i}`
  }
}
```